src/Symfony/Component/Validator/Constraints/FileValidator.php

```
@@ -143,7 +143,33 @@ public function validate(mixed $value, Constraint $constraint)
         $sizeInBytes = filesize($path);
         $basename = $value instanceof UploadedFile ? $value->getClientOriginalName() : basename($path);
 
-        if ($constraint->filenameMaxLength && $constraint->filenameMaxLength < $filenameLength = \strlen($basename)) {
+        try {
+            $invalidFilenameCharset = !@mb_check_encoding($basename, $constraint->filenameCharset);
+        } catch (\ValueError $e) {
+            if (!str_starts_with($e->getMessage(), 'mb_check_encoding(): Argument #2 ($encoding) must be a valid encoding')) {
+                throw $e;
+            }
+
+            $invalidFilenameCharset = true;
+        }
+
+        $filenameLength = $invalidFilenameCharset ? 0 : match ($constraint->filenameCountUnit) {
+            File::FILENAME_COUNT_BYTES => \strlen($basename),
+            File::FILENAME_COUNT_CODEPOINTS => mb_strlen($basename, $constraint->filenameCharset),
+            File::FILENAME_COUNT_GRAPHEMES => grapheme_strlen($basename),
+        };
```

Should we throw an exception if someone sends `'foo'` for example as `filenameCountUnit` ?

Can we also add a test for this? thanks

---

Yes, if someone sends 'foo', we throw an [exception](https://github.com/symfony/symfony/pull/58485/files#diff-738b9954d83ec3560d6a0b70f45958640c89abd268d72bba647e87c0ff13e185R177) at the constraint level, and it is also covered by the [test](https://github.com/symfony/symfony/pull/58485/files#diff-7a668456d6d5f71d2bd7300406f7f635e23f3a70a5995fc45e5017066755f776R101). thanks